prow/cmd/deck/static/common/common.ts

```diff
@@ -89,18 +89,21 @@ export namespace cell {
 		}
 	};
 
-	export function commitRevision(repo: string, ref: string, SHA: string): HTMLTableDataCellElement {
+	export function commitRevision(repo: string, ref: string, SHA: string, hostType: string, codeHost: string): HTMLTableDataCellElement {
 		const c = document.createElement("td");
 		const bl = document.createElement("a");
 		bl.href = "https://github.com/" + repo + "/commit/" + SHA;
+		if (hostType === "gerrit") {
+			bl.href = codeHost + "/" + repo + "/+/" + SHA;
```

Comment:
It seems strange to pass individually useless substrings of the URL that must be combined (in a manner dependent on the hostType) to create the actual usable URLs. Why not just pass the usable URLs directly and allow the Prow component that makes the ProwJob to specify the URLs at creation time? It doesn't make sense to have Deck be responsible for understanding the link format for different hosts. It looks like you really want to be exposing a review/pr link, a code/commit link, and an author link.

Reply:
> allow the Prow component that makes the ProwJob to specify the URLs at creation time

This sounds fine in theory, although in practice I don't know all the different places jobs are created, and I'm hesitant to duplicate a bunch of code or risk missing links somewhere. Will grepping for `NewProwJob` and `NewProwJobWithAnnotation` give me a comprehensive list of spots to look at, or are there other ways of creating jobs I should be aware of?

Reply:
Why would this necessitate code duplication? As for risking missing links, I don't think we need to be concerned with that if we change the appropriate function signatures to require the links as parameters.

FWIW grepping for those functions should yield all the places ProwJobs are created.

Reply:
Okay, my plan now is to:
1. Add links to `Refs` at job creation time.
2. Instead of doing another layer of parsing `Refs` in `deck/jobs` or thereabouts (this chain doesn't need that many links (no pun intended)), we'll just serve `Refs` directly to the client, and do any remaining necessary processing on the front end.

What do you think? cc @stevekuznetsov